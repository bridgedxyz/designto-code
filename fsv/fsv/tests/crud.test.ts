import { Volume } from "memfs";

const volume = Volume.fromJSON({});

// create a directory
test("mkdir", () => {
  volume.mkdirSync("/foo");

  console.log(volume);

  expect(volume.toJSON()).toMatchSnapshot();
});

// create a file
test("touch", () => {
  volume.writeFileSync("/foo/bar.txt", "hello world");

  console.log(volume);
  console.log(volume.fds[0]);
  console.log(volume.toJSON());

  expect(volume.toJSON()).toMatchSnapshot();
});

// move a file
// test("mv", () => {
//   volume.mkdirSync("/foo");
//   volume.writeFileSync("/foo/bar.txt", "hello world");
//   volume.renameSync("/foo/bar.txt", "/foo/baz.txt");

//   console.log(volume);

//   expect(volume.toJSON()).toMatchSnapshot();
// });
