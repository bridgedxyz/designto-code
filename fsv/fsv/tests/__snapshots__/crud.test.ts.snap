// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create 1`] = `
{
  "/foo/bar.txt": "hello world",
}
`;

exports[`mkdir 1`] = `
{
  "/foo": null,
}
`;

exports[`touch 1`] = `
{
  "/foo/bar.txt": "hello world",
}
`;
